source/d3d11/d3d11_device_context.hpp

```diff
@@ -183,22 +183,32 @@ struct D3D11DeviceContext : ID3D11DeviceContext3
 	virtual void STDMETHODCALLTYPE SetMarkerInt(LPCWSTR pLabel, INT Data) override;
 	virtual void STDMETHODCALLTYPE BeginEventInt(LPCWSTR pLabel, INT Data) override;
 	virtual void STDMETHODCALLTYPE EndEvent() override;
-	#pragma endregion
-	#pragma region ID3D11DeviceContext3
+#pragma endregion
+#pragma region ID3D11DeviceContext3
 	virtual void STDMETHODCALLTYPE Flush1(D3D11_CONTEXT_TYPE ContextType, HANDLE hEvent) override;
 	virtual void STDMETHODCALLTYPE SetHardwareProtectionState(BOOL HwProtectionEnable) override;
 	virtual void STDMETHODCALLTYPE GetHardwareProtectionState(BOOL *pHwProtectionEnable) override;
-	#pragma endregion
+#pragma endregion
 
 	void log_drawcall(UINT vertices);
+	ID3D11DepthStencilView *copy_depthstencil(ID3D11DepthStencilView *depthstencil);
+	com_ptr<ID3D11Texture2D> copy_depth_texture(D3D11_TEXTURE2D_DESC texture_desc, ID3D11Texture2D *depthtexture);
+	bool check_depthstencil(ID3D11DepthStencilView* pDepthStencilView, com_ptr<ID3D11Texture2D> texture, D3D11_TEXTURE2D_DESC texture_desc);
 	void set_active_depthstencil(ID3D11DepthStencilView* pDepthStencilView);
+	void set_active_OM_depthstencil(ID3D11DepthStencilView* pDepthStencilView);
+	void set_active_cleared_depthstencil(ID3D11DepthStencilView* pDepthStencilView);
 
 	void clear_drawcall_stats();
 
 	LONG _ref = 1;
+	unsigned int _vertices_treshold = 10000;
 	ID3D11DeviceContext *_orig;
 	unsigned int _interface_version;
 	D3D11Device *const _device;
 	com_ptr<ID3D11DepthStencilView> _active_depthstencil;
+	com_ptr<ID3D11Texture2D> _depth_texture;
 	reshade::d3d11::draw_call_tracker _draw_call_tracker;
+	static unsigned int _clear_DSV_iter;
```

Comment:
`static` variables are really bad in ReShade. Especially here! Many modern games spawn multiple device contexts on differents threads, so that the code using these variables would now potentially access them simultanously and cause all kinds of bad behavior (race conditions, ...).

Reply:
Yes, i know. Will add a mutex here (did it before), but in my testings, did not encountered any race conditions conflict. Leaved it like that in order nort to impact perfs.
Are you agree i add a mutex for it ?

Reply:
Why not use atomic variables? No need for a mutex. See for example http://en.cppreference.com/w/cpp/atomic/atomic.